### PR TITLE
NO-ISSUE - Fix vsphere controller not implementing all abstract methods.

### PR DESCRIPTION
**Pull request by eliorerz**
- Fix vsphere controller not implementing all abstract methods
- Rename vsphere controller to match the configurations naming convention

/cc @michaellevy101

### CODE REVIEW

File: src/tests/base_test.py [hunk 1]
```diff
@@ -82,7 +82,7 @@ def new_controller_configuration(self, request: FixtureRequest) -> BaseNodeConfi
         :rtype: new node controller configuration
         """
         if global_variables.platform == consts.Platforms.VSPHERE:
-            config = VSphereControllerConfig()
+            config = BaseVSphereConfig()
```
Comment (osherdp) [inline src/tests/base_test.py:85]: not sure I understand
if it's now abstract, how can we instantiate it?

**Review by osherdp — CHANGES_REQUESTED** [review]

File: src/tests/base_test.py (hunk 1 above)
Comment (eliorerz) [reply src/tests/base_test.py]: Right, missed that. Fixed

**Review by osherdp — APPROVED** [review]

**Pull request merged**